core/controllers/reader.py

```diff
@@ -124,15 +124,16 @@ def get(self, exploration_id):
         """Handles GET requests."""
         version_str = self.request.get('v')
         version = int(version_str) if version_str else None
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id, strict=False)
 
         # Note: this is an optional argument and will be None when the
         # exploration is being played outside the context of a collection.
         collection_id = self.request.get('collection_id')
         can_edit = (
```

Comment:
I think you can get rid of the outermost parens and do `can_edit = rights_manager.check_can_edit_activity(` on one line. Ditto below.